src/navigation/decorate.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+
+function isCurrentLink(item, currentUrl){
+	return (item.href && item.href === currentUrl) || (item.id && currentUrl.includes(item.id));
+}
+
+function decorateDrawerItem(item, currentUrl){
```

Comment:
Yeah, the drawer has a different structure to the others (nested arrays, headings and `item.item`) was easier to just split it out